File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int i=0;
        int j=numbers.length-1;

        int sum = 0;
        while(i < j) {
            sum = numbers[i] + numbers[j]; 
            if(sum == target) {
                if(i != j) {
                    if(i < j) {
                        return new int[]{i+1, j+1};
                    } else {
                        return new int[]{j+1, i+1};
                    }
                }
            } else if(sum < target) {
                i++;
            } else {
                j--;
            }
        }

        return new int[]{-1, -1};
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mymap = {}

        for i in range(0, len(nums)):
            num = nums[i]
            complement = target - num

            if complement in mymap:
                return [mymap[complement], i]
            
            mymap[num] = i
        
        return [-1, -1]
''' 
| Java HashMap     | Python dict              |
| ---------------- | ------------------------ |
| `put(k, v)`      | `d[k] = v`               |
| `get(k)`         | `d.get(k)` or `d[k]`     |
| `containsKey(k)` | `k in d`                 |
| `remove(k)`      | `del d[k]` or `d.pop(k)` |
| `size()`         | `len(d)`                 |
| `clear()`        | `d.clear()`              |
| `keySet()`       | `d.keys()`               |
| `values()`       | `d.values()`             |
| `entrySet()`     | `d.items()`              |
 '''